### PR TITLE
Update API REST specs to 7.2.0

### PR DESCRIPTION
**Pull request by codebrain**
- Update ApiGenerator
- Add stability field
- Exclude new APIs
- Add XML comments for Beta / Experimental stability

### CODE REVIEW

File: src/CodeGeneration/ApiGenerator/Generator/ApiGenerator.cs [hunk 1]
```diff
@@ -48,6 +48,15 @@ public static async Task Generate(string downloadBranch, params string[] folders
 				}
 			}
 
+			// Check if there are any non-Stable endpoints present.
+			foreach (var endpoint in spec.Endpoints)
+			{
+				if (endpoint.Value.Stability != Stability.Stable)
+				{
+					Warnings.Add($"Endpoint {endpoint.Value.Name} is not marked as Stable ({endpoint.Value.Stability})");
```
Comment (Mpdreamz) [inline src/CodeGeneration/ApiGenerator/Generator/ApiGenerator.cs]: :heart:

File: src/Nest/Requests.Rollup.cs [hunk 2]
```diff
@@ -106,7 +108,8 @@ Id Id
 		}
 	}
 
-	///<summary>Request for GetCapabilities <para></para></summary>
+	///<summary>Request for GetCapabilities</summary>
+	///<remarks>Note: Experimental within the Elasticsearch server, it is highly likely to break or be removed in the near future and has no backwards compatibility guarantees.</remarks>
```
Comment (Mpdreamz) [inline src/Nest/Requests.Rollup.cs]: :heart:

Comment (russcam) [reply src/Nest/Requests.Rollup.cs]: > it is highly likely to break or be removed in the near future

I'm not sure about this wording - is it highly likely? 

We could use the same wording as the reference docs:

> This functionality is experimental and may be changed or removed completely in a future release. Elastic will take a best effort approach to fix any issues, but experimental features are not subject to the support SLA of official GA features.

File: src/CodeGeneration/ApiGenerator/Domain/Specification/Stability.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+using System.Runtime.Serialization;
+
+namespace ApiGenerator.Domain.Specification {
```
Comment (russcam) [inline src/CodeGeneration/ApiGenerator/Domain/Specification/Stability.cs]: Minor nitpick: formatting

File: src/CodeGeneration/ApiGenerator/Views/HighLevel/Requests/RequestImplementations.cshtml [hunk 4]
```diff
@@ -5,14 +5,32 @@
...
 }
-	///<summary>Request for @names.MethodName <para>@r.OfficialDocumentationLink</para></summary>
+///<summary>Request for @names.MethodName@Raw(r.OfficialDocumentationLink.IsNullOrEmpty() ? "" : " <para>" + r.OfficialDocumentationLink + "</para>")</summary>
+@if (r.Stability != Stability.Stable)
+{
+	string warningMessage = "";
+	switch (r.Stability)
+	{
+		case Stability.Experimental:
+			warningMessage = "it is highly likely to break or be removed in the near future and has no backwards compatibility guarantees.";
```
Comment (russcam) [inline src/CodeGeneration/ApiGenerator/Views/HighLevel/Requests/RequestImplementations.cshtml]: I think we should use the same wording as the reference documentation for _experimental_:

> This functionality is experimental and may be changed or removed completely in a future release. Elastic will take a best effort approach to fix any issues, but experimental features are not subject to the support SLA of official GA features.

File: src/CodeGeneration/ApiGenerator/Views/HighLevel/Requests/RequestImplementations.cshtml [hunk 5]
```diff
@@ -5,14 +5,32 @@
...
+			break;
+		case Stability.Beta:
+			warningMessage = "it is less likely to break or be removed but still reserve the right to do so.";
```
Comment (russcam) [inline src/CodeGeneration/ApiGenerator/Views/HighLevel/Requests/RequestImplementations.cshtml]: I think we should use the same wording as the reference documentation for _beta_:

> This functionality is in beta and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.

**Pull request merged**